src/OrderInfo.java

```diff
@@ -1,12 +1,12 @@
 public class OrderInfo {
     public Order order;
-    public ObserverStates orderObserver;
+    public Observer orderObserver;
     public OrderState orderState;
 
     /** set new state of the order
      * @return true if the order is not ready, otherwise false */
     public boolean setNextState() {
-        orderState = orderObserver.stateUpdate(orderState);
+        orderState = orderState.next();
```

Comment:
You forgot to notify the observer
```suggestion
        orderState = orderState.next();
        if (orderState != null) {
            orderObserver.stateUpdate(orderState);
        }
```